src/engine.rs

```diff
@@ -47,6 +47,12 @@ pub struct Engine {
     has_history: bool,
 }
 
+pub enum Signal {
+    SUCCESS(String),
+    SIGINT, // Typically Ctrl-c
+    EOF, // Typically Ctrl-d
```

Comment:
I think you could use `CtrlC` and `CtrlD` here instead of SIGINT/EOF. Helps with the readability and probably is a little more accurate if we're running on a platform without signals.